mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractCombinationManagerTest.java

```diff
@@ -0,0 +1,360 @@
+        MessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
+
+        assertThat(messageManager2.getMetaData(true, session, FetchGroup.FIRST_UNSEEN).getMessageCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void getMetadataFromMessageManagerShouldReturnNumberOfUnseenMessageWhenSetInMailboxesFromMessageIdManager() throws Exception {
+        MessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
+
+        assertThat(messageManager2.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getUnseenCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void getMetadataFromMessageManagerShouldReturnFirstUnseenMessageWhenSetInMailboxesFromMessageIdManager() throws Exception {
+        ComposedMessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS);
+
+        messageIdManager.setInMailboxes(messageId.getMessageId(), ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
+
+        assertThat(messageManager2.getMetaData(true, session, FetchGroup.FIRST_UNSEEN).getFirstUnseen()).isEqualTo(messageId.getUid());
+    }
+
+    @Test
+    public void getMetadataFromMessageManagerShouldReturnNumberOfUnseenMessageWhenSetFlagsFromMessageIdManager() throws Exception {
+        Flags newFlag = new Flags(Flag.RECENT);
+        MessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
+
+        messageIdManager.setFlags(newFlag, FlagsUpdateMode.ADD, messageId, ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
+
+        assertThat(messageManager1.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getUnseenCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void getMetadataFromMessageManagerShouldReturnFirstUnseenMessageWhenSetFlagsFromMessageIdManager() throws Exception {
+        Flags newFlag = new Flags(Flag.USER);
+        ComposedMessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS);
+
+        messageIdManager.setFlags(newFlag, FlagsUpdateMode.ADD, messageId.getMessageId(), ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
+
+        assertThat(messageManager1.getMetaData(true, session, FetchGroup.FIRST_UNSEEN).getFirstUnseen()).isEqualTo(messageId.getUid());
+    }
```

Comment:
It seems you are missing some tests using setInMailboxes to move a message from a mailbox to an other one.